### PR TITLE
12.0 mig account invoice refund option

### PR DESCRIPTION
**Pull request by ajayjaviya**
Task : https://www.druidoo.io/web#id=794&action=361&active_id=28&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

File: account_invoice_refund_option/tests/test_account_invoice_refund_option.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from .common import AccountInvoiceRefundOptionTest
```
Comment (ivantodorovich) [inline account_invoice_refund_option/tests/test_account_invoice_refund_option.py:1]: Why a separate file for these tests? :thinking:

File: account_invoice_refund_option/tests/test_account_invoice_refund_option.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            Test the account invoice refund option to refund only selected
+            invoice line.
+        """
+        # Check invoice and invoice line
+        assert self.Invoice and self.Invoice.invoice_line_ids
+
+        # Create refund of selected invoice line
+        invoice_refund_wizard = self.account_invoice_refund_obj.create({
+            'date_invoice': datetime.today().strftime(
+                DEFAULT_SERVER_DATETIME_FORMAT),
```
Comment (ivantodorovich) [inline account_invoice_refund_option/tests/test_account_invoice_refund_option.py]: In odoo 12 date fields are python date objects. Why converting to string?
There's no need for `DEFAULT_SERVER_DATETIME_FORMAT`, is it?

File: account_invoice_refund_option/view/view_account_invoice_refund.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                groups="sale.group_discount_per_so_line" />
+                            <field name="invoice_line_tax_ids"
+                                widget="many2many_tags" />
+                            <field name="price_subtotal" />
+                        </tree>
+                    </field>
+                </group>
+            </xpath>
+        </field>
+    </record>
```
Comment (ivantodorovich) [inline account_invoice_refund_option/view/view_account_invoice_refund.xml:34]: FYI, there's no need to limit characters on xml lines.
No need to change anythnig here, but it seems you're trying to fit xml files in 80 character lines, and it's not necessary

File: account_invoice_refund_option/view/view_account_invoice_refund.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                            <field name="quantity" />
+                            <field name="uom_id"
+                                groups="uom.group_uom" />
+                            <field name="price_unit" />
+                            <field name="discount"
+                                groups="sale.group_discount_per_so_line" />
+                            <field name="invoice_line_tax_ids"
+                                widget="many2many_tags" />
+                            <field name="price_subtotal" />
+                        </tree>
```
Comment (ivantodorovich) [inline account_invoice_refund_option/view/view_account_invoice_refund.xml]: I'm guessing this is the only thing that's forcing us to depend on `sale` module, right?

If that's the case, can't we just reuse an existing view for invoice lines? or we don't have one?

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

File: account_invoice_refund_option/view/view_account_invoice_refund.xml (hunk 4 above)
Comment (ajayjaviya) [reply account_invoice_refund_option/view/view_account_invoice_refund.xml]: @ivantodorovich : Yes it due to sale group , I have remove it and use same as of invoice line

File: account_invoice_refund_option/tests/test_account_invoice_refund_option.py (hunk 1 above)
Comment (ajayjaviya) [reply account_invoice_refund_option/tests/test_account_invoice_refund_option.py:1]: @ivantodorovich : I keep in separate file to define static variables from actual test case

File: account_invoice_refund_option/tests/test_account_invoice_refund_option.py (hunk 2 above)
Comment (ajayjaviya) [reply account_invoice_refund_option/tests/test_account_invoice_refund_option.py]: @ivantodorovich  : yes use default date object

**Pull request merged**